amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/ConfigsBuilderTest.java

```diff
@@ -53,12 +54,27 @@ public class ConfigsBuilderTest {
     private static final String WORKER_IDENTIFIER = "worker-id";
 
     @Test
-    public void testTrackerConstruction() {
+    public void testTrackerConstructionByStreamName() {
         final String streamName = "single-stream";
         final ConfigsBuilder configByName = createConfig(streamName);
         final ConfigsBuilder configBySingleTracker = createConfig(new SingleStreamTracker(streamName));
 
-        for (final ConfigsBuilder cb : Arrays.asList(configByName, configBySingleTracker)) {
+        testTrackerConstruction(streamName, configBySingleTracker, configByName);
+    }
+
+    @Test
+    public void testTrackerConstructionByStreamARN() {
+        final Arn streamARN = Arn.fromString("arn:aws:kinesis:region:123456789012:stream/stream-name");
+        final ConfigsBuilder configByArn = createConfig(streamARN);
+        final ConfigsBuilder configBySingleTracker = createConfig(new SingleStreamTracker(streamARN));
+
+        final String streamName = streamARN.resource().resource();
```

Comment:
Suggestion: for test legibility, include a `assertEquals("stream-name", streamName);`